### ISSUE TITLE
Use HTML5 input types

### ISSUE THREAD

**Issue by srsgores** [jakkor/ember-advanced-form#1]
Helps with mobile browsers, providing correct keyboard.

For example, 
https://github.com/jakkor/ember-advanced-form/blob/master/addon/templates/components/integer.hbs#L4 should be:

```
{{input type="number"}}
```

Likewise, for time:

```
{{input type="time"}}
```

**Comment by jakkor** [comment]
I wanted to use html5 style but saw that not all browsers have support for it. The plan is to check if it's available and use it. Definitely will add that in future.

**Comment by srsgores** [comment]
Yes, but if you at least change the `type`, then the browsers that don't support it will gracefully fall back to `text`, so why not?

**Comment by srsgores** [comment]
@jakkor, I can do the feature detection.  How about a PR?

**Comment by jakkor** [comment]
Yes, of course. Help will be very much appreciated.
BTW the field name is integer because I wanted to have separate integer and float fields in future.

Didn't know about changing to text in older browser, so maybe this will be enough ?
